pkgs/native_assets_cli/lib/src/locking/locking.dart

```diff
@@ -50,23 +50,26 @@ Future<T> _runUnderFileLock<T>(
   Duration? timeout,
   Logger? logger,
 }) async {
-  await _ensureLockfileExists(file, logger);
-
-  var timedOut = false;
-  Timer? timeoutTimer;
-  if (timeout != null) {
-    timeoutTimer = Timer(timeout, () {
-      timedOut = true;
-    });
-  }
-
+  if (!await file.exists()) await file.create(recursive: true);
   final randomAccessFile = await file.open(mode: FileMode.write);
-  var locked = false;
   var printed = false;
-  while (!locked && !timedOut) {
+  final stopwatch = Stopwatch()..start();
+  while (timeout == null || stopwatch.elapsed < timeout) {
     try {
       await randomAccessFile.lock(FileLock.exclusive);
-      locked = true;
+      try {
+        if (!Platform.isWindows) {
+          // On Windows, a file can't be written to once we have a lock on it.
+          // https://github.com/dart-lang/sdk/issues/56378
+          await file.writeAsString(
+            'Last acquired by ${Platform.resolvedExecutable} '
+            'running ${Platform.script} on ${DateTime.now()}.',
```

Comment:
You may want to include `Process.pid`: If someone sees the lock file and looks at contents, to see whether it is still used, one would want to look whether the process is still running, processes are identified by PIDs

Reply:
Great idea!